### ISSUE TITLE
Astro 2.1.4 breaks build

### ISSUE THREAD

**Issue by leesei** [satnaing/astro-paper#48]
`@astrojs/react/client.js` in `node_modules/@astrojs/react/src/index.ts` gets resolved to `./src/astrojs/react/client.js` by `tsconfig.json`

```sh
 yarn build
yarn run v1.22.19
$ astro build && jampack ./dist
node:fs:600
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open '/home/leesei/blog/astro-paper/src/astrojs/react/client.js'
    at Object.openSync (node:fs:600:3)
    at Object.readFileSync (node:fs:468:35)
    at extractExportsData (file:///home/leesei/blog/astro-paper/node_modules/astro/node_modules/vite/dist/node/chunks/dep-79892de8.js:45100:31) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/leesei/blog/astro-paper/src/astrojs/react/client.js'
}

Node.js v18.14.2
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I tired adding two paths `"@*": ["node_modules/@*", "./src/*"]` according to https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping but then the theme's code fails.

**Comment by leesei** [comment]
The workaround is the `yarn add astro@2.1.3` for now.

**Comment by satnaing** [comment]
It seems to be the problem with the latest release, version 2.1.4, since v2.1.3 and others are working fine. 
I'll try to search for potential solutions to resolve this problem. 
If not found, might as well open an issue on Astro repo.

The best way to avoid this problem for now is to stick with astro@2.1.3 and not upgrade to astro@2.1.4.

**Comment by satnaing** [comment]
Fixed in #49
